mvpa2/tests/test_rsa.py

```diff
@@ -73,6 +76,53 @@ def test_PDistConsistency():
     assert_array_almost_equal(res4.samples,cres2)
 
 
+def test_CDist():
+    targets = np.tile(range(3), 2)
+    chunks = np.repeat(np.array((0,1)), 3)
+    ds = dataset_wizard(samples=data, targets=targets, chunks=chunks)
+    train_data = ds[ds.sa.chunks == 0, ]
+    test_data = ds[ds.sa.chunks == 1, ]
+
+    # Some distance metrics
+    metrics = ['euclidean', 'correlation', 'cityblock', 'mahalanobis']
+    VI_mahalanobis = np.eye(5)
+    for sattr in [['targets'], None]:
+        for metric in metrics:
+            metric_kwargs = {'VI': VI_mahalanobis} if metric == 'mahalanobis' \
+                else {}
+            scipy_cdist = cdist(train_data.samples, test_data.samples,
+                        metric, **metric_kwargs)
+            pymvpa_cdist = CDist(pairwise_metric=metric,
+                        pairwise_metric_kwargs=metric_kwargs,
+                        sattr=sattr)
+
+            assert_true(not pymvpa_cdist.is_trained)
+            pymvpa_cdist.train(train_data)
+            assert_true(pymvpa_cdist.is_trained)
+            res = pymvpa_cdist(test_data)
+            # Check to make sure the cdist results are close to CDist results
+            assert_array_almost_equal(res.samples.ravel(),
+                                      scipy_cdist.ravel())
```

Comment:
since to be trained once and then reused on new data -- would have been nice to test that it provides correct result (comparing to independent new train/test cycle)